File: 0563-binary-tree-tilt/0563-binary-tree-tilt.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int ans = 0;
    int func(TreeNode* root){
        if(root == NULL){
            return 0;
        }
        int leftVal = func(root->left);
        int rightVal = func(root->right);
        ans += abs(leftVal - rightVal);        
        return root->val + leftVal + rightVal;
    }
    int findTilt(TreeNode* root) {
        int n = func(root);
        return ans;
    }
};

File: 0563-binary-tree-tilt/0563-binary-tree-tilt.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int FindTilt(TreeNode root) {
        int tilt = 0;
        postOrder(root);
        return tilt;

        int postOrder(TreeNode root){
            if(root == null){
                return 0;
            }

            int left = postOrder(root.left);
            int right = postOrder(root.right);

            tilt += Math.Abs(left - right);
            return root.val + left + right;
        }
    }
}